core/src/main/scala/codesearch/core/index/repository/Extensions.scala

```diff
@@ -3,25 +3,29 @@ package codesearch.core.index.repository
 import simulacrum.typeclass
 
 @typeclass trait Extensions[A] {
-  def extensions: Set[String]
+  def commonExtensions: Set[String] = Set("json", "md", "txt", "xml", "yml", "yaml", "properties")
```

Comment:
it would also be nice to keep some extension-less files, e.g. "readme", "changelog", "changes", "Makefile"

even better -- detect small text-only files and keep them no matter what, but I guess we can do that later